storage/s3.go

```diff
@@ -794,14 +794,15 @@ func (sc *SessionCache) newSession(ctx context.Context, opts Options) (*session.
 	if opts.NoVerifySSL {
 		httpClient = insecureHTTPClient
 	}
-
+	// WithDisableRestProtocolURICleaning is added to allow adjacent slashes to be used in s3 object keys.
```

Comment:
Could be moved right above the `WithDisableRestProtocolURICleaning` call.
```suggestion
	// Disable URI cleaning to allow adjacent slashes to be used in S3 object keys.
```